src/v/cluster/security_frontend.cc

```diff
@@ -255,18 +267,31 @@ ss::future<std::vector<delete_acls_result>> security_frontend::do_delete_acls(
      */
     auto removed_bindings = _authorizer.local().remove_bindings(filters, true);
 
-    delete_acls_cmd_data data;
-    data.filters = std::move(filters);
-    delete_acls_cmd cmd(std::move(data), 0 /* unused */);
+    auto should_sanction
+      = _features.local().should_sanction()
+        && absl::c_any_of(
+          removed_bindings, [](const std::vector<security::acl_binding>& bs) {
+              return absl::c_any_of(bs, [](const security::acl_binding& b) {
+                  return b.entry().principal().type()
+                         == security::principal_type::role;
+              });
+          });
 
     errc err;
-    try {
-        auto ec = co_await replicate_and_wait(
-          _stm, _as, std::move(cmd), model::timeout_clock::now() + timeout);
-        err = map_errc(ec);
-    } catch (const std::exception& e) {
-        vlog(clusterlog.warn, "Unable to delete ACLs: {}", e);
-        err = errc::replication_error;
+    if (should_sanction) {
+        err = errc::feature_disabled;
```

Comment:
suggestion: Maybe a warn level log here as well

Reply:
added some logs here and to the any_of operation...kinda clunky but lmk what you think.